src/reconcile/rec_visibility.c

```diff
@@ -83,11 +91,14 @@ __rec_append_orig_value(
         /*
          * Done if the on page value already appears on the update list. We can't do the same check
          * for stop time point because we may still need to append the onpage value if only the
-         * tombstone is on the update chain.
+         * tombstone is on the update chain. We only need to check the in memory case as in other
+         * cases, the update must have been restored from data store.
          */
         if (unpack->tw.start_ts == upd->start_ts && unpack->tw.start_txn == upd->txnid &&
-          upd->type != WT_UPDATE_TOMBSTONE)
+          upd->type != WT_UPDATE_TOMBSTONE) {
+            WT_ASSERT(session, F_ISSET(S2C(session), WT_CONN_IN_MEMORY));
```

Comment:
I didn't clearly understand the reason behind this assert. Can you explain it?

Reply:
For the non-inmem cases, it is covered by checking WT_UPDATE_RESTORED_FROM_DS flag.